### DISCUSSION TITLE
How do I keep the spawned tab open when the command has finished?

### DISCUSSION THREAD

**Discussion by 1dancook** [wezterm/wezterm#5988] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

I had this ALT+c keybind that I used in tmux to open a floating window with `cal -3` which shows the previous, current, and next month calendar. It's handy and fast.

I'm trying to implement this in wezterm, but the opened tab closes instantly (because the process has finished).

So, how can I make the tab stay open?

Here is my key binding:

```
{ mods = 'OPT', key = 'c', action = wezterm.action.SpawnCommandInNewTab {args = {"/usr/bin/cal", "-3"}} }
```

I have tried to add `; read` in a variety of ways to no avail.

I couldn't find any built-in option with the wezterm action. Perhaps I missed something?

**Comment by 1dancook** [comment]
This appears to be a different issue with spawning commands in general; closing this now.

**Comment by wez** [comment, marked as answer]
Take a look at https://wezfurlong.org/wezterm/config/lua/config/exit_behavior.html

**Reply by 1dancook** [reply]
Thank you for this! I missed this in the config somehow. 

After trying a few approaches with that, I settled on keeping the default behavior and pointing the spawn command to execute a local script:

weztermcal
```bash
#!/usr/bin/env bash

# A simple script to use for wezterm to print the calendar
# with a keyboard shortcut, i.e. ALT+c

cal -3
false # keep the window open with the default behavior of CloseOnCleanExit
```

The reason I took this approach is because I think that I'd like to keep the default behavior, but at the same time I couldn't get `cal -3; false` working within _args_ for a spawn command. I tried all manner of variations and splits. 

Initially I thought that including `exit_behavior` as an argument for spawn commands would help with this kind of thing, however, I don't know if it really makes sense to do that; I may revisit that idea after using wezterm for a while. 

Again, thank you for your time!
